### ISSUE TITLE
Improved custom background options

### ISSUE THREAD

**Issue by Hebgbs** [dubtrack/www-dubtrack-fm#169]
tl;dr: More first-party options for community background image would be nice, with additional tweaking depending on how much effort a creator wants to put into everything appearing "Right" on a variety of displays.

Dubtrack creators should retain the option of using a single image across all display sizes, but give advanced options for creators who really care about it to define images for specific resolutions. A simplified version of this to not completely overwhelm creators might be to allow for an image per display aspect ratio, so creators wouldn't need to make or modify an image for every display size, but both can co-exist as a toggle to control how much automation is done by Dubtrack, as "Single image", "Simple display sizes" and "Advanced display sizes";

Single image: As Dubtrack does right now, limited by as big as the image allows to fill the screen. In this mode, creators only have access to putting a single image up, as well, scaling options, which would be "Tiled", "Stretch", "Fill", "Fit" and "Original size";
.
Stretch: 100% X and Y
Fill: 100% X, Y = X
Fit: 100% Y, X = Y
Tiled: Repeat image by X and Y
Original size: Use as-is without sizing or repeat.

Also present should be an Image placement specification that would allow a creator to define where the image "Begins" for a user, which would be constant across all displays.

Simple display sizes: Show only aspect ratios, and specify an image as default when a standard aspect ratio is unavailable. For simplicity, force creators to stick with a single image scaling and image anchor option.

Advanced display sizes: Show every resolution, while also allowing an image per aspect ratio when a user signs on using an unspecified display size. Image specifications from simple display sizes would be inherited as advanced aspect ratio defaults.

For HD sizes, allowing specific simple / advanced aspect ratio default settings for 4K and 8K would be nice so when a user needs to render a 16:9 image, it does not appear super-pixelated on those larger resolutions. This, of course doesn't matter if the image is a small tiled GIF, but the option for a creator to define different configurations of the same image for different displays would not only improve a community's brand identity, but also improve the aesthetics of communities overall when users are signing in from a variety of devices, from phones and tablets with ever-changing orientations to the user with that terribly-expensive quad-SLI, triple-display GPU config.

Interface implementation for all of the above would be simple enough; A toggle for single image, simple resolutions or advanced resolutions would be presented to the end user, for the latter two, have radio buttons so a user can select a default image when a user joins with an undefined aspect ratio or viewable display size, then when a user joins, do some simple maths to figure out the aspect ratio when either simple mode is in use, or when a display size of the same available aspect ratio but not of the same display size is used.

Specifying the placement of an image could be done with two toggles, three options each; Left, Centre, Right, and Top, Centre, Bottom.

Even though the above had been assuming DubTrack grabs the current display resolution, there are toolbars, sidebars and other guff that stands between a user and taking full advantage of their available viewable space. DubTrack should be using only the nearest available display size, which would allow creators to test their settings by scaling their browser to change the viewable space.

**Comment by swordling** [comment]
I like this idea! Only issue I can see overall is it being a little to complicated for new users. But that could be easily solved by adding an "Advanced options" toggle or something similar. :+1:

**Comment by Hebgbs** [comment]
Kind of as I said earlier. Simple mode is aspect ratio only which would would affect _all_ resolutions and use an image which is nearest to the aspect ratio of the viewable browser window,

I realize I was mentioning both the browser grabbing the resolution of one's current display and using the browser's viewable size, in hindsight people have toolbars and nothing is truly exact, but if there is a way to detect the browser window being maximized, it could be specified that the image changes from viewable size to display size. This would be great for most rigs, but as a Firefox user using Tile View, and for people using Always On Top in Chrome, that can get a little wonky.

So for simplicity, redacting what I said about resolution and relying exclusively on viewable size would be the better choice to target everybody. I will modify the above initial report to reflect this.

**Comment by avatarkava** [comment]
Wouldn't ability to set a custom CSS block (here we could use media queries in CSS3) handle all this?  

It seems like this would be a more flexible option than trying to account for every permutation under the sun in an 'advanced' tab.

**Comment by Hebgbs** [comment]
That could be so, but couldn't we also have Javascript that has blanks a user can fill in via the frontend? It's not like I am asking for the world, just a means for creators to make their background (and ultimately, the community's initial identity for a first-time visitor) seamless between devices.

**Comment by RobGoldy** [comment]
I think this overcomplicates the user interface if you add a custom Javascript and CSS block for input. The settings tab for room already is complicated as-is. Only the host or co hosts of the room should have access to seeing it btw.
